design/ceph/node-loss-rbd-rwo.md

```diff
@@ -0,0 +1,108 @@
+
+- Remove the volume attachment from the lost node
+- Delete the old pod on the lost node
+- Create a new pod on the new node
+- Allow the volume to be attached to the new node
+
+Once this taint is applied manually, rook will create a [NetworkFence CR](https://github.com/csi-addons/kubernetes-csi-addons/blob/main/docs/networkfence.md). The [csi-addons operator](https://github.com/csi-addons/kubernetes-csi-addons) will then blocklist the node to prevent any ceph rbd client on the lost node from writing any more data.
+
+After the new pod is running on the new node and the old node which was lost comes back, rook will delete the [NetworkFencr CR](https://github.com/csi-addons/kubernetes-csi-addons/blob/main/docs/networkfence.md).
+
+example of taint to be applied to lost node:
+
+```console
+kubectl taint nodes <node-name> node.kubernetes.io/out-of-service=nodeshutdown:NoExecute
+# or
+kubectl taint nodes <node-name> node.kubernetes.io/out-of-service=nodeshutdown:NoSchedule
+```
+
+> Note: This will be enabled by default in rook, in the case for some reason user wants to disable this feature in rook can edit `rook-ceph-operator-config` configmap and update the `ROOK_WATCH_FOR_NODE_FAILURE: "false"`.
+
+## How get which NodeIp to block
+
+There are multiple networking options available for example, host Networking, Pod networking, Multus etc. This make it difficult to know which NodeIp address to blockList.
+For this we'll follow the following approach which will work for all networking options(but still will not work for external cluster which nodeIps):
+
+1. Get the `volumesInUse` from the node which has the taint `out-of-service`
+
+2. List all the pv and compare with value of `volumesInUse`
+
+3. For Ceph volumes on that node:
+
+   1. If RBD PVC makes use of the rbd status API
+      1. example:
+
+    ```console
+    $ rbd status <image_name>
+    Watchers:
+    watcher=172.21.12.201:0/4225036114 client.17881 cookie=18446462598732840961
+    ```
+
+   2. If CephFS PVC uses ceph fs client ls  API
+      1.
+        ```Will add example for Cephfs```
+
+4. Get IPs from step 3 (in above example `172.21.12.201:0/4225036114`)
+5. Blocklist the entire ip with port.
+
+Example of a NetworkFence CR that the Rook operator would create when a node is lost:
+
+```yaml
+apiVersion: csiaddons.openshift.io/v1alpha1
+kind: NetworkFence
+metadata:
+  name: <name> # We will keep the name the same as the node name
+  namespace: <ceph-cluster-namespace>
+spec:
+  driver: <driver-name> # for us it will `rook-ceph.rbd.csi.ceph.com`
+  fenceState: <fence-state> # For us it will be `Fenced`
+  cidrs:
+    - <172.21.12.201:0/4225036114> # from above section, how to get node ip to block
```

Comment:
Step 5 above says "Blocklist the entire ip with port". This looks like it is just blocklisting the specific CIDRs. So we aren't blocklisting the entire IP, right? Should it say "Blocklist the endpoint where the volumes are found connected"?

Reply:
changed to `Blocklist the endpoint where the volumes are  mounted`